### PR TITLE
Printable Model browser

### PR DESCRIPTION
**Pull request by zahnno**
PR includes:
- Model browser which displays UML diagram & properties of Models available through nSpecDAO.

### CODE REVIEW

File: src/foam/doc/DocBrowser.js [hunk 1]
```diff
@@ -76,6 +76,20 @@ foam.CLASS({
       name: 'axiom',
       hidden: true
     },
+    {
+      name: 'documentation',
+      label: 'Documentation',
+      tableCellFormatter: function(value, obj, axiom) {
```
Comment (kgrgreer) [inline src/foam/doc/DocBrowser.js]: How is this different than what happens by default?

File: src/foam/doc/ModelBrowser.js [hunk 2]
```diff
@@ -0,0 +1,238 @@
+foam.CLASS({
+  package: 'foam.doc',
+  name: 'ModelBrowser',
+  extends: 'foam.u2.Element',
+  documentation: 'Show UML & properties for passed in models',
+
+  requires: [
+    'foam.doc.DocBorder',
```
Comment (kgrgreer) [inline src/foam/doc/ModelBrowser.js]: Sort lists.

File: src/foam/doc/ModelBrowser.js [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+    }
+  `,
+
+  methods: [
+    function initE(){
+      this.SUPER();
+      var self = this;
+
+      this.nSpecDAO.select().then(function(d){
+        d.array.forEach(function(n){
```
Comment (kgrgreer) [inline src/foam/doc/ModelBrowser.js]: You could just pass this function to select() directly and skip the extra step/nesting.

File: src/foam/doc/ModelBrowser.js [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+          if ( cls.class ) {
+            self.models.push(cls.class)
+            self.models = Array.from(self.models)
+          }
+        });
+      });
+      this.start().addClass(this.myClass())
+        .start('h2').add("Model Browser").end()
+        .start().add(this.PRINT_PAGE).end()
+        .add(this.slot(function(models) {
```
Comment (kgrgreer) [inline src/foam/doc/ModelBrowser.js]: Do you need to use slot()? Is models going to change?

Comment (zahnno) [reply src/foam/doc/ModelBrowser.js]: How can I grab my models before initE method?

**Pull request merged**